packages/aws-cdk-lib/core/test/feature-flag-report.test.ts

```diff
@@ -0,0 +1,75 @@
+import { FeatureFlagReportProperties } from '@aws-cdk/cloud-assembly-schema';
+import * as cxapi from '../../cx-api';
+import { App } from '../lib';
+import { generateFeatureFlagReport } from '../lib/private/feature-flag-report';
+
+jest.mock('../../cx-api', () => {
+  const actual = jest.requireActual('../../cx-api');
+  return {
+    ...actual,
+    CloudAssemblyBuilder: jest.fn().mockImplementation(() => ({
+      addArtifact: jest.fn(),
+      buildAssembly: jest.fn().mockReturnValue({
+        tryGetArtifact: jest.fn().mockReturnValue('aws-cdk-lib/feature-flag-report'),
+      }),
+    })),
+  };
+});
+
+describe('generate feature flag report', () => {
+  test('feature flag report can be retrieved from CloudAssembly using its artifact ID', () => {
+    const app = new App();
+    const builder = new cxapi.CloudAssemblyBuilder('/tmp/test');
+
+    generateFeatureFlagReport(builder, app);
```

Comment:
I would do an actual `app.synth()` and test that the right information gets written to the assembly.

Don't mock the builder.